lambda-buffers-frontend/src/LambdaBuffers/Frontend/Parsec.hs

```diff
@@ -0,0 +1,194 @@
+module LambdaBuffers.Frontend.Parsec (parseModule, parseImport, runParser) where
+
+import Control.Applicative (Alternative ((<|>)))
+import Control.Monad (MonadPlus (mzero), void)
+import Data.Kind (Type)
+import Data.Maybe (isJust)
+import Data.String (IsString (fromString))
+import Data.Text (Text)
+import LambdaBuffers.Frontend.Syntax (ConstrName (ConstrName), Constructor (Constructor), Import (Import), Module (Module), ModuleAlias (ModuleAlias), ModuleName (ModuleName), ModuleNamePart (ModuleNamePart), Product (Product), SourceInfo (SourceInfo), SourcePos (SourcePos), Ty (TyApp, TyRef', TyVar), TyArg (TyArg), TyBody (Opaque, Sum), TyDef (TyDef), TyName (TyName), TyRef (TyRef), VarName (VarName))
+import Text.Parsec (ParseError, ParsecT, SourceName, Stream, alphaNum, char, endOfLine, eof, getPosition, label, lower, many, many1, optionMaybe, optional, runParserT, sepBy, sepEndBy, sourceColumn, sourceLine, sourceName, space, string, try)
+import Text.Parsec.Char (upper)
+
+type Parser :: Type -> (Type -> Type) -> Type -> Type
+type Parser s m a = ParsecT s () m a
+
+runParser :: (Stream s IO Char) => Parser s IO a -> SourceName -> s -> IO (Either ParseError a)
+runParser p = runParserT (p <* eof) ()
+
+parseUpperCamelCase :: Stream s m Char => Parser s m Text
```

Comment:
random: I tend to prepend with `p` - `pUpperCamelCase`. This is completely fine as well.

Reply:
Feeling lazy, but definitely noted. Next refactor I do that